tests/std/tests/GH_004109_iter_value_t_direct_initialization/test.compile.pass.cpp

```diff
@@ -0,0 +1,700 @@
+
+    // Partitioning operations
+    (void) ranges::is_partitioned(nil, nil, pred);
+    ranges::partition(nil, nil, pred);
+    ranges::partition_copy(nil, nil, nil, nil, pred);
+    ranges::stable_partition(nil, nil, pred);
+    (void) ranges::partition_point(nil, nil, pred);
+
+    // Sorting operations
+    (void) ranges::is_sorted(nil, nil);
+    (void) ranges::is_sorted(nil, nil, comp);
+
+    (void) ranges::is_sorted_until(nil, nil);
+    (void) ranges::is_sorted_until(nil, nil, comp);
+
+    ranges::sort(nil, nil);
+    ranges::sort(nil, nil, comp);
+
+    ranges::partial_sort(nil, nil, nil);
+    ranges::partial_sort(nil, nil, nil, comp);
+
+    ranges::partial_sort_copy(nil, nil, nil, nil);
+    ranges::partial_sort_copy(nil, nil, nil, nil, comp);
+
+    ranges::stable_sort(nil, nil);
+    ranges::stable_sort(nil, nil, comp);
+
+    ranges::nth_element(nil, nil, nil);
+
+    // Binary search operations (on sorted ranges)
+    (void) ranges::lower_bound(nil, nil, val);
+    (void) ranges::lower_bound(nil, nil, val, comp);
+
+    (void) ranges::upper_bound(nil, nil, val);
+    (void) ranges::upper_bound(nil, nil, val, comp);
+
+    (void) ranges::binary_search(nil, nil, val);
+    (void) ranges::binary_search(nil, nil, val, comp);
+
+    (void) ranges::equal_range(nil, nil, val);
+    (void) ranges::equal_range(nil, nil, val, comp);
+
+    // Set operations (on sorted ranges)
+    ranges::merge(nil, nil, nil, nil, nil);
+    ranges::merge(nil, nil, nil, nil, nil, comp);
+
+    ranges::inplace_merge(nil, nil, nil);
+    ranges::inplace_merge(nil, nil, nil, comp);
+
+    (void) ranges::includes(nil, nil, nil, nil);
+    (void) ranges::includes(nil, nil, nil, nil, comp);
+
+    ranges::set_difference(nil, nil, nil, nil, nil);
+    ranges::set_difference(nil, nil, nil, nil, nil, comp);
+
+    ranges::set_intersection(nil, nil, nil, nil, nil);
+    ranges::set_intersection(nil, nil, nil, nil, nil, comp);
+
+    ranges::set_symmetric_difference(nil, nil, nil, nil, nil);
+    ranges::set_symmetric_difference(nil, nil, nil, nil, nil, comp);
+
+    ranges::set_union(nil, nil, nil, nil, nil);
+    ranges::set_union(nil, nil, nil, nil, nil, comp);
+
+    // Heap operations
+    (void) ranges::is_heap(nil, nil);
+    (void) ranges::is_heap(nil, nil, comp);
+
+    (void) ranges::is_heap_until(nil, nil);
+    (void) ranges::is_heap_until(nil, nil, comp);
+
+    ranges::make_heap(nil, nil);
+    ranges::make_heap(nil, nil, comp);
+
+    ranges::push_heap(nil, nil);
+    ranges::push_heap(nil, nil, comp);
+
+    ranges::pop_heap(nil, nil);
+    ranges::pop_heap(nil, nil, comp);
+
+    ranges::sort_heap(nil, nil);
+    ranges::sort_heap(nil, nil, comp);
+
+    // Minimum/maximum operations
+    (void) ranges::max_element(nil, nil);
+    (void) ranges::max_element(nil, nil, comp);
+
+    (void) ranges::min_element(nil, nil);
+    (void) ranges::min_element(nil, nil, comp);
+
+    //(void) ranges::minmax_element(nil, nil);

```

Comment:
Perhaps we should pass 5 `nil`s to select the following overload.
https://github.com/microsoft/STL/blob/0403d19f5461fd15983737c3f01ec34800ea9275/stl/inc/algorithm#L3517-L3521

Reply:
I reduced some code.
```c++
#include <utility>

struct I {
};

template <class _Ty>
    struct min_max_result {
        _Ty min;
        _Ty max;

        template <typename _Ty2>
        constexpr operator min_max_result<_Ty2>() && {
            return {std::move(min), std::move(max)};
        }
    };

auto test() {
  I i;
  min_max_result<I> t {i, i};
  return t; // warning C4700: uninitialized local variable 't' used
}
```
I guess CL don't like the two `std::move` usages for I. Other compilers seems OK with that.
https://godbolt.org/z/Yddss868e

Reply:
If I add the constructors then we don't have the error:
```C++
struct I {
    I() = default;
    I(I&&) = default;
    I(const I&) = default;
};
```
If I remove `I(I&&) = default;` then I also got `warning C4700: uninitialized local variable 't' used`.
So I guess CL generates `I(const I&) = default;` by itself and not the `    I(I&&) = default;`
But gcc seems generate this `inline constexpr I(I &&) noexcept = default;`  constructor instead of `I(const I&) = default;`.
Look at: https://cppinsights.io/s/947f4c20

Should I create a bug for CL? It seems it should create and use the move constructor instead of the copy one.
I found 
```
    8 If the definition of a class X does not explicitly declare a move constructor, a non-explicit one will be implicitly declared as defaulted if and only if
        (8.1) — X does not have a user-declared copy constructor,
        (8.2) — X does not have a user-declared copy assignment operator,
        (8.3) — X does not have a user-declared move assignment operator, and
        (8.4) — X does not have a user-declared destructor.
```

from https://stackoverflow.com/questions/4943958/conditions-for-automatic-generation-of-default-copy-move-ctor-and-copy-move-assi  and I recheck the current draft too: https://eel.is/c++draft/class.copy.ctor#8